src/UKHO.S100PermitService.Common/Clients/IProductKeyServiceApiClient.cs

```diff
@@ -0,0 +1,7 @@
+﻿namespace UKHO.S100PermitService.Common.Clients
+{
+    public interface IProductKeyServiceApiClient
+    {
+        Task<HttpResponseMessage> CallProductKeyServiceApiAsync(string uri, HttpMethod httpMethod, string payLoad, string accessToken, string correlationId);
```

Comment:
This method signature isn't right, it breaks encapsulation in all sorts of ways - please see comments on the client implementation.

Reply:
Resolved